internal/build/pipeline_state.go

```diff
@@ -35,7 +46,7 @@ const buildStepOutputPrefix = "     "
 
 func NewPipelineState(ctx context.Context, totalStepCount int, c Clock) *PipelineState {
 	return &PipelineState{
-		curPipelineStep:        1,
+		curPipelineStep:        PipelineStep{Index: 0},
```

Comment:
IIUC it's idiomatic go to omit initializations to 0-values like this (both the `Index` and the `PipelineStep` here are such initializations)

(this is irrelevant if you agree with my below comment to delete `curPipelineStep`)